OpenRA.Game/Graphics/HardwarePalette.cs

```diff
@@ -79,6 +79,9 @@ public void AddPalette(string name, ImmutablePalette p, bool allowModifiers)
 				CopyPaletteToBuffer(index, p);
 		}
 
+		[System.Diagnostics.CodeAnalysis.SuppressMessage(
+			"Performance", "CA1854:Prefer the 'IDictionary.TryGetValue(TKey, out TValue)' method",
+			Justification = "False positive - indexer is a set not a get.")]
```

Comment:
I wonder if we can silence this warning by writing
```
		public void ReplacePalette(string name, IPalette p)
		{
			if (mutablePalettes.ContainsKey(name))
			{
				var palette = new MutablePalette(p);
				mutablePalettes[name] = palette;
				CopyPaletteToBuffer(indices[name], palette);
			}
			else if (palettes.ContainsKey(name))
			{
				var palette = new ImmutablePalette(p);
				palettes[name] = palette;
				CopyPaletteToBuffer(indices[name], palette);
			}
			else
				throw new InvalidOperationException($"Palette `{name}` does not exist");

			CopyBufferToTexture();
		}
```

Reply:
This does not prevent it from incorrectly raising the warning.